lib/ansible/modules/cloud/vmware/vmware_network_protocol_profile.py

```diff
@@ -0,0 +1,244 @@
+        else:
+            # for backwards-compat
+            return None
+
+    def process_state(self):
+        try:
+            ippool_states = {
+                'absent': {
+                    'present': self.state_remove_ippool,
+                    'absent': self.state_exit_unchanged,
+                },
+                'present': {
+                    'present': self.state_exit_unchanged,
+                    'absent': self.state_add_ippool,
+                }
+            }
+
+            ippool_states[self.state][self.check_ippool_state()]()
+
+        except vmodl.RuntimeFault as runtime_fault:
+            self.module.fail_json(msg=runtime_fault.msg)
+        except vmodl.MethodFault as method_fault:
+            self.module.fail_json(msg=method_fault.msg)
+        except Exception as e:
+            self.module.fail_json(msg=str(e))
+
+    def state_exit_unchanged(self):
+        self.module.exit_json(changed=False)
+
+    def state_remove_ippool(self):
+        changed = True
+        result = None
+        try:
+            self.ip_pool_obj = self.select_ip_pool()
+            self.content.ipPoolManager.DestroyIpPool(self.dc_obj, self.ip_pool_obj.id, False)
+            self.module.exit_json(changed=changed)
+        except vim.fault.InvalidState as invalid_state:
+            self.module.fail_json(msg="Unable to delete Ip Pool with name %s" % self.ip_pool)
+
+    def state_add_ippool(self):
+        changed = True
+        if self.dc_obj is None:
+            self.module.fail_json(msg="Unable to find datacenter with name %s" % self.datacenter)
+        if self.dvswitch_obj is None:
+            self.module.fail_json(msg="Unable to find distributed switch with name %s" % self.dvswitch)
+        self.dvsportgrp_obj = find_dvspg_by_name(self.dvswitch_obj, self.network)
+
+        IpPool_assoc = vim.vApp.IpPool.Association()
+        IpPool_assoc.network = self.dvsportgrp_obj
+        IpPool_assoc.networkName = self.network
+
+        IpPool_config = vim.vApp.IpPool.IpPoolConfigInfo()
+        IpPool_config.dhcpServerAvailable = self.dhcpserverAvailable
+        IpPool_config.gateway = self.gateway
+        IpPool_config.ipPoolEnabled = self.ipPoolEnabled
+        IpPool_config.dns = self.dns
+        IpPool_config.subnetAddress = self.subnetAddress
+        IpPool_config.netmask = self.netmask
+
+        IpPool_data = vim.vApp.IpPool()
+        IpPool_data.name = self.ip_pool
+        IpPool_data.ipv4Config = IpPool_config
+        IpPool_data.dnsDomain = self.dnsDomain
+        IpPool_data.dnsSearchPath = self.dnsSearchPath
+        IpPool_data.networkAssociation = [IpPool_assoc]
+
+        self.content.ipPoolManager.CreateIpPool(self.dc_obj, IpPool_data)
+
+        self.module.exit_json(changed=changed)
+
+    def check_ippool_state(self):
+
+        ip_pools = self.content.ipPoolManager.QueryIpPools(self.dc_obj)
+        for i in ip_pools:
+            if i.name == self.ip_pool:
+                self.ip_pool_obj = i
```

Comment:
What happens if we have multiple IP pool with same name. In the past, we have seen a lot of issues due to the same name objects.